tests/pipelines/test_api.py

```diff
@@ -0,0 +1,194 @@
+
+# pylint:disable=redefined-outer-name
+# pylint:disable=unused-argument
+
+import copy
+import mock
+import pytest
+
+import databricks_cli.pipelines.api as api
+from databricks_cli.pipelines.api import LibraryObject
+
+PIPELINE_ID = '123456'
+SPEC = {
+    'id': PIPELINE_ID,
+    'name': 'test_pipeline'
+}
+CREDENTIALS = 'dummy_credentials'
+HEADERS = 'dummy_headers'
+
+
+@pytest.fixture()
+def pipelines_api():
+    with mock.patch('databricks_cli.pipelines.api.DeltaPipelinesService') \
+            as DeltaPipelinesServiceMock:
+        DeltaPipelinesServiceMock.return_value = mock.MagicMock()
+        _pipelines_api = api.PipelinesApi(None)
+        yield _pipelines_api
+
+
+def test_partition_correctly_identifies_local_paths(pipelines_api):
+    libraries = [
+        LibraryObject('jar', '//absolute/path/abc.ext'),
+        LibraryObject('jar', '/relative/path.ext'),
+        LibraryObject('jar', 'file://file/scheme/abs/path.ext'),
+        LibraryObject('jar', 'file:/file/scheme/relative/path.ext'),
+        LibraryObject('jar', 'FILE:/all/caps.ext'),
+        LibraryObject('jar', 'FiLe:/weird/case.ext'),
+        LibraryObject('jar', 'file.ext')
+    ]
+    expected_llo = [
+        LibraryObject('jar', '//absolute/path/abc.ext'),
+        LibraryObject('jar', '/relative/path.ext'),
+        LibraryObject('jar', '//file/scheme/abs/path.ext'),
+        LibraryObject('jar', '/file/scheme/relative/path.ext'),
+        LibraryObject('jar', '/all/caps.ext'),
+        LibraryObject('jar', '/weird/case.ext'),
+        LibraryObject('jar', 'file.ext')
+    ]
+    expected_rest = []
+    llo, rest = pipelines_api._partition_libraries_and_extract_local_paths(libraries)
+    assert llo == expected_llo
+    assert rest == expected_rest
+
+
+def test_partition_correctly_identifies_remote_paths(pipelines_api):
+    libraries = [
+        LibraryObject('jar', 'dbfs:/absolute/path/abc.ext'),
+        LibraryObject('jar', 's3:file://file/scheme/abs/path.ext'),
+        LibraryObject('jar', 's3:file:/file/scheme/relative/path.ext'),
+        LibraryObject('jar', 'scheme:file.ext'),
+        LibraryObject('jar', 'scheme://abs/path.ext'),
+        LibraryObject('jar', 'scheme:/rel/path.ext'),
+        LibraryObject('egg', 'file:/rel/path.ext'),
+        LibraryObject('whl', '/rel/path.ext')
+    ]
+    expected_llo = []
+    expected_rest = [
+        LibraryObject('jar', 'dbfs:/absolute/path/abc.ext'),
+        LibraryObject('jar', 's3:file://file/scheme/abs/path.ext'),
+        LibraryObject('jar', 's3:file:/file/scheme/relative/path.ext'),
+        LibraryObject('jar', 'scheme:file.ext'),
+        LibraryObject('jar', 'scheme://abs/path.ext'),
+        LibraryObject('jar', 'scheme:/rel/path.ext'),
+        LibraryObject('egg', 'file:/rel/path.ext'),
+        LibraryObject('whl', '/rel/path.ext')
+    ]
+    llo, rest = pipelines_api._partition_libraries_and_extract_local_paths(libraries)
+    assert llo == expected_llo
+    assert rest == expected_rest
```

Comment:
What are this test and the one above testing that you're not testing in test_deploy? the only thing is different schemes and relative paths, right? But here you're never actually testing the ability to use relative paths. I'd make test_deploy take in the list of input paths (like below you have [{'jar': jar1}, {'jar': jar2}, {'jar': 'dbfs:/pipelines/code/file.jar'}]) and the expected output paths and verify these that way.